### PR TITLE
Refactored class_optimize.py

### PR DESCRIPTION
**Pull request by NormannK**
- Optimized Imports: Removed unused imports and organized them.
- Refactored Code: Introduced split_individual function for clarity.
- Improved Efficiency: Enhanced penalty calculation and streamlined loops.
- Updated Evaluation Logic: Better handling of penalties in evaluate.
- Type Hints added.

Notes:
- Fixed seed option added for automated tests
- Verbose option added, default False
- isfloat is only used in flask_server.py
- start_hour is not used in this class
- identical results with the old version using test.py and fixed seeds

### CODE REVIEW

**Comment by NormannK** [comment]
statistically proof that the new class is as good as (the same) as the old one.
image
see #93 for results from the original class

File: modules/class_optimize.py [hunk 1]
```diff
@@ -373,121 +282,71 @@ def optimierung_ems(
...
             eautocharge_hours_float,
-            temperature_forecast,
+            parameter["temperature_forecast"],
             start_hour,
             self.prediction_hours,
             einspeiseverguetung_euro_pro_wh,
             extra_data=extra_data,
         )
+
         os.system("cp visualisierungsergebnisse.pdf ~/")
```
Comment (fortrieb) [inline modules/class_optimize.py:341]: Result PDF path file should be configurable, maybe a method parameter or config parameter.

File: modules/class_optimize.py [hunk 2]
```diff
@@ -1,307 +1,233 @@
...
+            if attr in creator.__dict__:
+                del creator.__dict__[attr]
 
+        # Create new FitnessMin and Individual classes
         creator.create("FitnessMin", base.Fitness, weights=(-1.0,))
         creator.create("Individual", list, fitness=creator.FitnessMin)
 
-        # PARAMETER
+        # Initialize toolbox with attributes and operations
         self.toolbox = base.Toolbox()
```
Comment (fortrieb) [inline modules/class_optimize.py:88]: Initialization should be move to `__init__`. Toolbox is used in other places as well.

File: modules/class_optimize.py [hunk 3]
```diff
@@ -1,307 +1,233 @@
...
-        self.prediction_hours = prediction_hours  #
+    def __init__(
+        self,
+        prediction_hours: int = 24,
+        strafe: float = 10,
+        optimization_hours: int = 24,
+        verbose: bool = False,
+        fixed_seed: Optional[int] = None,
+    ):
+        """Initialize the optimization problem with the required parameters."""
```
Comment (fortrieb) [inline modules/class_optimize.py:37]: Docstring can be more helpful to describe each parameter. I'm using [Google Style docstrings](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings), maybe we can discuss on using in this project.

Comment (drbacke) [reply modules/class_optimize.py:37]: Lets discuss this in the discussion section.

**Pull request merged**